### ISSUE TITLE
SslStream CopyToAsync_AllDataCopied_Large test failing in CI

### ISSUE THREAD

**Issue by wfurt** [dotnet/runtime#44175]
I only see failures win various Linux flavors. 65 failures in last 30 days

```
    System.Net.Security.Tests.SslStreamNetworkConformanceTests.CopyToAsync_AllDataCopied_Large(useAsync: True) [FAIL]
      System.IO.IOException : Unable to read data from the transport connection: Connection reset by peer.
      ---- System.Net.Sockets.SocketException : Connection reset by peer
      Stack Trace:
        /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs(781,0): at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](TIOAdapter adapter, Memory`1 buffer)
        /_/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs(96,0): at System.IO.Stream.<CopyToAsync>g__Core|29_0(Stream source, Stream destination, Int32 bufferSize, CancellationToken cancellationToken)
        /_/src/libraries/Common/tests/Tests/System/IO/StreamConformanceTests.cs(1400,0): at System.IO.Tests.ConnectedStreamConformanceTests.CopyToAsync_AllDataCopied(Int32 byteCount, Boolean useAsync)
        /_/src/libraries/Common/tests/Tests/System/IO/StreamConformanceTests.cs(1375,0): at System.IO.Tests.ConnectedStreamConformanceTests.CopyToAsync_AllDataCopied_Large(Boolean useAsync)
        --- End of stack trace from previous location ---
        ----- Inner Stack Trace -----
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs(959,0): at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.CreateException(SocketError error, Boolean forAsyncThrow)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs(719,0): at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ReceiveAsync(Socket socket, CancellationToken cancellationToken)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs(227,0): at System.Net.Sockets.Socket.ReceiveAsync(Memory`1 buffer, SocketFlags socketFlags, Boolean fromNetworkStream, CancellationToken cancellationToken)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs(575,0): at System.Net.Sockets.NetworkStream.ReadAsync(Memory`1 buffer, CancellationToken cancellationToken)
        /_/src/libraries/System.Net.Security/src/System/Net/Security/ReadWriteAdapter.cs(34,0): at System.Net.Security.AsyncReadWriteAdapter.ReadAsync(Memory`1 buffer)
        /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs(781,0): at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](TIOAdapter adapter, Memory`1 buffer)
        /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncMethodBuilderCore.cs(42,0): at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
           at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](TIOAdapter adapter, Memory`1 buffer)
        /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.cs(821,0): at System.Net.Security.SslStream.ReadAsync(Memory`1 buffer, CancellationToken cancellationToken)
        /_/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs(96,0): at System.IO.Stream.<CopyToAsync>g__Core|29_0(Stream source, Stream destination, Int32 bufferSize, CancellationToken cancellationToken)
        /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs(286,0): at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s)
        /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs(186,0): at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
        /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs(324,0): at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)
        /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs(302,0): at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext()
        /_/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskContinuation.cs(795,0): at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox box, Boolean allowInlining)
        /_/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs(3219,0): at System.Threading.Tasks.Task.RunContinuations(Object continuationObject)
        /_/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Future.cs(400,0): at System.Threading.Tasks.Task`1.TrySetResult(TResult result)
        /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs(438,0): at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetExistingTaskResult(Task`1 task, TResult result)
        /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs(895,0): at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](TIOAdapter adapter, Memory`1 buffer)
        /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs(186,0): at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
        /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs(324,0): at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread)
        /_/src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.cs(1076,0): at System.Threading.ThreadPool.<>c.<.cctor>b__82_0(Object state)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs(891,0): at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.InvokeContinuation(Action`1 continuation, Object state, Boolean forceAsync, Boolean requiresExecutionContextFlow)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs(685,0): at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.OnCompleted(SocketAsyncEventArgs _)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs(207,0): at System.Net.Sockets.SocketAsyncEventArgs.OnCompletedInternal()
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs(979,0): at System.Net.Sockets.SocketAsyncEventArgs.FinishOperationAsyncSuccess(Int32 bytesTransferred, SocketFlags flags)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs(383,0): at System.Net.Sockets.SocketAsyncEventArgs.CompletionCallback(Int32 bytesTransferred, SocketFlags flags, SocketError socketError)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs(110,0): at System.Net.Sockets.SocketAsyncEventArgs.TransferCompletionCallbackCore(Int32 bytesTransferred, Byte[] socketAddress, Int32 socketAddressSize, SocketFlags receivedFlags, SocketError socketError)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs(501,0): at System.Net.Sockets.SocketAsyncContext.BufferMemoryReceiveOperation.InvokeCallback(Boolean allowPooling)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs(931,0): at System.Net.Sockets.SocketAsyncContext.OperationQueue`1.ProcessAsyncOperation(TOperation op)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs(1345,0): at System.Net.Sockets.SocketAsyncContext.ProcessAsyncReadOperation(ReadOperation op)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs(338,0): at System.Net.Sockets.SocketAsyncContext.ReadOperation.System.Threading.IThreadPoolWorkItem.Execute()
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs(286,0): at System.Net.Sockets.SocketAsyncContext.AsyncOperation.Process()
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs(908,0): at System.Net.Sockets.SocketAsyncContext.OperationQueue`1.ProcessSyncEventOrGetAsyncEvent(SocketAsyncContext context, Boolean skipAsyncEvents, Boolean processAsyncEvents)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs(2096,0): at System.Net.Sockets.SocketAsyncContext.HandleEvents(SocketEvents events)
        /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs(248,0): at System.Net.Sockets.SocketAsyncEngine.System.Threading.IThreadPoolWorkItem.Execute()
        /_/src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.cs(612,0): at System.Threading.ThreadPoolWorkQueue.Dispatch()
        /_/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WorkerThread.cs(54,0): at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
        /_/src/coreclr/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs(43,0): at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
        /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs(186,0): at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
        /_/src/coreclr/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs(92,0): at System.Threading.ThreadHelper.ThreadStart()
        --- End of stack trace from previous location ---
```

**Comment by stephentoub** [comment]
The test here looks correct to me.

There are several variations of this test:
- Over NetworkStream vs over an in-memory stream
- useAsync == false or true

The failure you pasted above is with NetworkStream and useAsync:true.  Are all the failures like that?

**Comment by wfurt** [comment]
It fails for both sync & async. Here is example of sync failure (job 6b10b2b8-6142-4144-b32a-d3216276635b)
```
System.IO.IOException : Unable to read data from the transport connection: Connection reset by peer.\n---- System.Net.Sockets.SocketException : Connection reset by peer
   at System.Net.Sockets.NetworkStream.Read(Span`1 buffer) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs:line 257
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](TIOAdapter adapter, Memory`1 buffer)
   at System.Net.Security.SslStream.Read(Byte[] buffer, Int32 offset, Int32 count) in /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.cs:line 755
   at System.IO.Stream.CopyTo(Stream destination, Int32 bufferSize) in /_/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs:line 61
   at System.IO.Stream.CopyTo(Stream destination) in /_/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs:line 45
   at System.IO.Tests.ConnectedStreamConformanceTests.<>c__DisplayClass42_0.<CopyToAsync_AllDataCopied>b__0() in /_/src/libraries/Common/tests/Tests/System/IO/StreamConformanceTests.cs:line 1395
   at System.Threading.Tasks.Task.InnerInvoke() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:line 2371
   at System.Threading.Tasks.Task.<>c.<.cctor>b__278_0(Object obj) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:line 2359
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs:line 301
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:line 2321
--- End of stack trace from previous location ---
   at System.IO.Tests.ConnectedStreamConformanceTests.CopyToAsync_AllDataCopied(Int32 byteCount, Boolean useAsync) in /_/src/libraries/Common/tests/Tests/System/IO/StreamConformanceTests.cs:line 1400
   at System.IO.Tests.ConnectedStreamConformanceTests.CopyToAsync_AllDataCopied_Large(Boolean useAsync) in /_/src/libraries/Common/tests/Tests/System/IO/StreamConformanceTests.cs:line 1375
--- End of stack trace from previous location ---
----- Inner Stack Trace -----
```

Since I had local repro, I'll try to instrument little bit and get packet capture from the run. 
By quick glance, this fails only on newer distributions e.g. OpenSSL 1.1.1 -> TLS.13
There _may_ be some behavior differences in closing TLS.

**Comment by antonfirsov** [comment]
Is `System.Net.Security.Tests.SslStreamTls13NetworkConformanceTests.CopyToAsync_AllDataCopied_Large(useAsync: False)` related?

https://dev.azure.com/dnceng/public/_build/results?buildId=877008&view=ms.vss-test-web.build-test-results-tab
(under `net6.0-Linux-Debug-x64-CoreCLR_release-Ubuntu.1804.Amd64.Open`)

**Comment by stephentoub** [comment]
Yes. We split the test by TLS version to help confirm the theory this is a TLS 1.3-specific product bug.

**Comment by wfurt** [comment]
It seems like since 11/5 only the TLS13 variant is failing. ( I was originally confused with 16.04 queue name, but that is where we run containers with other OSes)  I will take look and I'll try to reproduce locally.

**Comment by wfurt** [comment]
I'm not sure if this is related but in my local run I found test hanging as well
```
'System.Net.Security.Tests.SslStreamMemoryConformanceTests.Read_Eof_Returns0', Elapsed: 15:34:16
  [Long Running Test] 'System.Net.Security.Tests.SslStreamTls12NetworkConformanceTests.Read_Eof_Returns0', Elapsed: 15:34:14
  [Long Running Test] 'System.Net.Security.Tests.SslStreamTls13NetworkConformanceTests.Read_Eof_Returns0', Elapsed: 15:34:12
     System.Net.Security.Tests: [Long Running Test] 'System.Net.Security.Tests.SslStreamMemoryConformanceTests.Read_Eof_Returns0', Elapsed: 15:36:16
  [Long Running Test] 'System.Net.Security.Tests.SslStreamTls12NetworkConformanceTests.Read_Eof_Returns0', Elapsed: 15:36:14
  [Long Running Test] 'System.Net.Security.Tests.SslStreamTls13NetworkConformanceTests.Read_Eof_Returns0', Elapsed: 15:36:12
```

In this particular case both TLS12 and TLS13 are in play.

**Comment by v-haren** [comment]
failed again in job: [runtime-libraries-coreclr outerloop 20201117.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=888901&view=ms.vss-test-web.build-test-results-tab&runId=28426066&resultId=102839&paneView=debug)

failed test: 
System.Net.Security.Tests.SslStreamTls13NetworkConformanceTests.CopyToAsync_AllDataCopied_Large(useAsync: True)
System.Net.Security.Tests.SslStreamTls13NetworkConformanceTests.CopyToAsync_AllDataCopied_Large(useAsync: False)

net6.0-Linux-Release-arm64-CoreCLR_release-(Alpine.312.Arm64.Open)ubuntu.1804.armarch.open@mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.12-helix-arm64v8-20200602002604-25f8a3e


Error message
~~~
System.IO.IOException : Unable to read data from the transport connection: Connection reset by peer.
---- System.Net.Sockets.SocketException : Connection reset by peer


Stack trace
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](TIOAdapter adapter, Memory`1 buffer) in /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs:line 781
   at System.IO.Stream.<CopyToAsync>g__Core|29_0(Stream source, Stream destination, Int32 bufferSize, CancellationToken cancellationToken) in /_/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs:line 102
   at System.IO.Tests.ConnectedStreamConformanceTests.CopyToAsync_AllDataCopied(Int32 byteCount, Boolean useAsync) in /_/src/libraries/Common/tests/Tests/System/IO/StreamConformanceTests.cs:line 2368
   at System.IO.Tests.ConnectedStreamConformanceTests.CopyToAsync_AllDataCopied_Large(Boolean useAsync) in /_/src/libraries/Common/tests/Tests/System/IO/StreamConformanceTests.cs:line 2343
--- End of stack trace from previous location ---
----- Inner Stack Trace -----
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.CreateException(SocketError error, Boolean forAsyncThrow) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs:line 959
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ReceiveAsync(Socket socket, CancellationToken cancellationToken) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs:line 719
   at System.Net.Sockets.Socket.ReceiveAsync(Memory`1 buffer, SocketFlags socketFlags, Boolean fromNetworkStream, CancellationToken cancellationToken) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs:line 227
   at System.Net.Sockets.NetworkStream.ReadAsync(Memory`1 buffer, CancellationToken cancellationToken) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/NetworkStream.cs:line 550
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](TIOAdapter adapter, Memory`1 buffer) in /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs:line 781
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncMethodBuilderCore.cs:line 42
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](TIOAdapter adapter, Memory`1 buffer)
   at System.Net.Security.SslStream.ReadAsync(Memory`1 buffer, CancellationToken cancellationToken) in /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.cs:line 821
   at System.IO.Stream.<CopyToAsync>g__Core|29_0(Stream source, Stream destination, Int32 bufferSize, CancellationToken cancellationToken) in /_/src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs:line 102
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 286
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs:line 186
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 324
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext() in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 302
   at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(IAsyncStateMachineBox box, Boolean allowInlining) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/TaskContinuation.cs:line 795
   at System.Threading.Tasks.Task.RunContinuations(Object continuationObject) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Task.cs:line 3228
   at System.Threading.Tasks.Task`1.TrySetResult(TResult result) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/Tasks/Future.cs:line 400
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetExistingTaskResult(Task`1 task, TResult result) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 438
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](TIOAdapter adapter, Memory`1 buffer) in /_/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs:line 895
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.ExecutionContextCallback(Object s) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 286
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs:line 186
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext(Thread threadPoolThread) in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 324
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.AsyncStateMachineBox`1.MoveNext() in /_/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncTaskMethodBuilderT.cs:line 302
   at System.Threading.ThreadPool.<>c.<.cctor>b__82_0(Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.cs:line 1076
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.InvokeContinuation(Action`1 continuation, Object state, Boolean forceAsync, Boolean requiresExecutionContextFlow) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs:line 891
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.OnCompleted(SocketAsyncEventArgs _) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Tasks.cs:line 685
   at System.Net.Sockets.SocketAsyncEventArgs.OnCompletedInternal() in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs:line 207
   at System.Net.Sockets.SocketAsyncEventArgs.FinishOperationAsyncSuccess(Int32 bytesTransferred, SocketFlags flags) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs:line 979
   at System.Net.Sockets.SocketAsyncEventArgs.CompletionCallback(Int32 bytesTransferred, SocketFlags flags, SocketError socketError) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs:line 383
   at System.Net.Sockets.SocketAsyncEventArgs.TransferCompletionCallbackCore(Int32 bytesTransferred, Byte[] socketAddress, Int32 socketAddressSize, SocketFlags receivedFlags, SocketError socketError) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs:line 110
   at System.Net.Sockets.SocketAsyncContext.BufferMemoryReceiveOperation.InvokeCallback(Boolean allowPooling) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs:line 501
   at System.Net.Sockets.SocketAsyncContext.OperationQueue`1.ProcessAsyncOperation(TOperation op) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs:line 931
   at System.Net.Sockets.SocketAsyncContext.ProcessAsyncReadOperation(ReadOperation op) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs:line 1345
   at System.Net.Sockets.SocketAsyncContext.ReadOperation.System.Threading.IThreadPoolWorkItem.Execute() in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs:line 338
   at System.Net.Sockets.SocketAsyncContext.AsyncOperation.Process() in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs:line 286
   at System.Net.Sockets.SocketAsyncContext.OperationQueue`1.ProcessSyncEventOrGetAsyncEvent(SocketAsyncContext context, Boolean skipAsyncEvents, Boolean processAsyncEvents) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs:line 908
   at System.Net.Sockets.SocketAsyncContext.HandleEvents(SocketEvents events) in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs:line 2096
   at System.Net.Sockets.SocketAsyncEngine.System.Threading.IThreadPoolWorkItem.Execute() in /_/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs:line 244
   at System.Threading.ThreadPoolWorkQueue.Dispatch() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.cs:line 705
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart() in /_/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WorkerThread.cs:line 54
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state) in /_/src/coreclr/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs:line 43
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) in /_/src/libraries/System.Private.CoreLib/src/System/Threading/ExecutionContext.cs:line 186
   at System.Threading.ThreadHelper.ThreadStart() in /_/src/coreclr/src/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs:line 92
--- End of stack trace from previous location ---
~~~

**Comment by wfurt** [comment]
fixed by #44741
